### DISCUSSION TITLE
Typescript error accessing `this`

### DISCUSSION THREAD

**Discussion by kinoli** [vuejs/pinia#1151] [Help and Questions] [answered]
Anytime I try to access `this` to reference the state or a getter I get an error.

```
Object is possibly 'undefined'. ts(2532)
```

I've followed the docs to the tee as far as I can see. Wondering what the solution to this is? 

Also, the docs could benefit from a Typescript section explaining various things.

<img width="503" alt="Screen Shot 2022-03-16 at 8 44 18 AM" src="https://user-images.githubusercontent.com/61486/158621016-cdcaf7f5-6596-4a4b-894a-f9adf04be72d.png">

**Comment by posva** [comment, marked as answer]
you need to use a normal function instead:

```ts
getters: {
  details(state) {
    this.
  }
}
```

Arrow functions bind their context to the context they are defined at: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

**Reply by nguyendvwanosoft** [reply]
thanks bro @posva

**Reply by elisechant** [reply]
life saver!
